tools/@o3r/test-helpers/src/create-test-app.ts

```diff
@@ -0,0 +1,135 @@
+  };
+
+  // Prepare folder
+  if (existsSync(appFolderPath)) {
+    rmSync(appFolderPath, {recursive: true});
+  }
+
+  if (options.blank) {
+    mkdirSync(appFolderPath, {recursive: true});
+    if (options.packageManager === 'yarn') {
+      execSync('yarn init', execAppOptions);
+    }
+  } else {
+    // Create app with ng new
+    const createOptions = `--directory=${options.appName} --style=scss --routing --defaults=true --skip-git --package-manager=${options.packageManager}`;
+    execSync(`${options.packageManager} create @angular@${options.angularVersion || ''} test-app ${options.packageManager === 'npm' ? '-- ' : ''}${createOptions}`,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      {cwd: options.cwd, stdio: 'inherit', env: {...process.env, NODE_OPTIONS: ''}});
+  }
+
+  if (options.packageManager === 'yarn') {
+    // Set yarn version
+    execSync('yarn config set enableStrictSsl false', execAppOptions);
+    if (options.yarnVersion) {
+      execSync(`yarn set version ${options.yarnVersion}`, execAppOptions);
+    }
+
+    // Set config to target local registry
+    execSync('yarn config set checksumBehavior update', execAppOptions);
+    execSync('yarn config set enableGlobalCache true', execAppOptions);
+    execSync('yarn config set enableImmutableInstalls false', execAppOptions);
+    execSync(`yarn config set globalFolder ${options.globalFolderPath}`, execAppOptions);
+    execSync('yarn config set nodeLinker pnp', execAppOptions);
+    execSync(`yarn config set npmScopes.ama-sdk.npmRegistryServer ${options.registry}`, execAppOptions);
+    execSync(`yarn config set npmScopes.o3r.npmRegistryServer ${options.registry}`, execAppOptions);
+    execSync('yarn config set unsafeHttpWhitelist localhost', execAppOptions);
+  } else {
+    // FIXME to be removed?
+    execSync('npm config set legacy-peer-deps=true -L project', execAppOptions);
+
+    execSync(`npm config set @ama-sdk:registry=${options.registry} -L project`, execAppOptions);
+    execSync(`npm config set @o3r:registry=${options.registry} -L project`, execAppOptions);
+    execSync(`npm config set cache=${options.globalFolderPath} -L project`, execAppOptions);
+  }
+
+  if (!options.blank) {
+    // Add dependencies
+    const deps = {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '@angular-devkit/schematics': options.angularVersion,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '@angular/pwa': options.angularVersion,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '@angular/material': options.materialVersion,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '@angular-devkit/core': options.angularVersion,
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      '@schematics/angular': options.angularVersion
+    };
+    const addCmd = options.packageManager === 'yarn' ? 'yarn add' : 'npm i';
```

Comment:
Cleaner to use the full command:
`yarn add`, `npm install`, `yarn exec ng` and `npm exec ng`

Reply:
updated